### PR TITLE
Update k8s/ocp versions used in e2e tests

### PR DESCRIPTION
**Pull request by naemono**
This is updating our e2e matrix to use the newest version of Kubernetes, and Openshift during e2e testing. I'm explicitly not updating the supported versions yet as we are nearing the next release.

### CODE REVIEW

**Comment by naemono** [comment]
buildkite test this -f p=ocp,TESTS_MATCH=TestSmoke -m s=8.11.3

**Comment by naemono** [comment]
buildkite test this -f p=kind,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31,TESTS_MATCH=TestSmoke -m s=8.11.3

**Comment by naemono** [comment]
buildkite test this -f p=kind,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.29.0@sha256:54a50c9354f11ce0aa56a85d2cacb1b950f85eab3fe1caf988826d1f89bf37eb,TESTS_MATCH=TestSmoke -m s=8.11.3

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 1]
```diff
@@ -23,14 +23,13 @@
...
     E2E_PROVIDER: kind
   mixed:
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.22.17@sha256:f5b2e5698c6c9d6d0adc419c0deae21a425c07d81bbf3b6a6834042f25d4fba2
-    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.23.17@sha256:59c989ff8a517a93127d4a536e7014d28e235fb3529d9fba91b3951d461edfdb
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.24.15@sha256:7db4f8bea3e14b82d12e044e25e34bd53754b7f2b0e9d56df21774e6f66a70ab
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.25.11@sha256:227fa11ce74ea76a0474eeefb84cb75d8dad1b08638371ecf0e86259b35be0c8
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.26.6@sha256:6e2d8b28a5b601defe327b98bd1c2d1930b49e5d8c512e1895099e4504007adb
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.27.3@sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
     - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
```
Comment (thbkrkr) [inline .buildkite/e2e/nightly-main-matrix.yaml:31]: We  an update the last 2 entries (ipv4 (default ip family) and ipv6):
```suggestion
    - DEPLOYER_KIND_NODE_IMAGE: kindest/node:v1.29.0@sha256:54a50c9354f11ce0aa56a85d2cacb1b950f85eab3fe1caf988826d1f89bf37eb
```

Comment (barkbay) [reply .buildkite/e2e/nightly-main-matrix.yaml:31]: Feeling like déjà vu 🤔 , could we add a comment to emphasize that we should test the latest version on both IPv4 and IPv6, and so the last K8s version must be referenced twice at the end of this list?

Comment (naemono) [reply .buildkite/e2e/nightly-main-matrix.yaml]: Comment added here, along with the latest kind/k8s twice.

File: .buildkite/e2e/nightly-main-matrix.yaml [hunk 2]
```diff
@@ -51,11 +52,10 @@
   fixed:
     E2E_PROVIDER: ocp
   mixed:
-    #- DEPLOYER_CLIENT_VERSION: "4.8.57"
-    #- DEPLOYER_CLIENT_VERSION: "4.9.58"
     #- DEPLOYER_CLIENT_VERSION: "4.10.55"
     #- DEPLOYER_CLIENT_VERSION: "4.11.33"
-    - DEPLOYER_CLIENT_VERSION: "4.13.4"
+    #- DEPLOYER_CLIENT_VERSION: "4.13.4"
+    - DEPLOYER_CLIENT_VERSION: "4.14.7"
```
Comment (barkbay) [inline .buildkite/e2e/nightly-main-matrix.yaml:58]: Last stable version advised for https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable-4.14/ is `4.14.6`, published on `Tue Dec 19 12:53:46 2023` . There is a version `4.14.7` but it seems to have been released 4 days before: https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/4.14.7/ 🤷

Comment (naemono) [reply .buildkite/e2e/nightly-main-matrix.yaml:58]: Well, I just went here and looked at the most recent by version:
https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/
But I do see your point @barkbay . I tested [yesterday](https://github.com/elastic/cloud-on-k8s/pull/7387#issuecomment-1860736813) with `4.14.7` successfully. Any preference on which version to move forward with? I have no strong opinion either way.

Comment (barkbay) [reply .buildkite/e2e/nightly-main-matrix.yaml:58]: I download the last client from RedHat:

Comment (barkbay) [reply .buildkite/e2e/nightly-main-matrix.yaml:58]: I downloaded the most recent version from the RedHat web site:

```
oc version
Client Version: 4.14.6
Kustomize Version: v5.0.1
```

I would go with `4.14.6` something looks wrong with `4.14.7`

**Comment by naemono** [comment]
buildkite test this -f p=ocp,TESTS_MATCH=TestSmoke -m s=8.11.3

File: .buildkite/e2e/nightly-main-matrix.yaml (hunk 2 above)
Comment (naemono) [reply .buildkite/e2e/nightly-main-matrix.yaml]: Sounds good. Testing now.

**Review by barkbay — APPROVED** [review]
LGTM

**Review by thbkrkr — APPROVED** [review]

**Pull request merged**